Jenkinsfile

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ * Includes the third-party code listed at http://links.sonatype.com/products/nexus/attributions.
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+@Library('ci-pipeline-library') _
+import com.sonatype.jenkins.pipeline.GitHub
+import com.sonatype.jenkins.pipeline.OsTools
+
+properties([
+  parameters([
+    booleanParam(defaultValue: false, description: 'Force Red Hat Certified Build for a non-master branch', name: 'force_red_hat_build'),
+    booleanParam(defaultValue: false, description: 'Skip Red Hat Certified Build', name: 'skip_red_hat_build'),
+  ])
+])
+
+node('ubuntu-zion') {
```

Comment:
why a legacy pipeline instead of a declarative pipeline? only asking.

Reply:
I copied this jenkinsfile from the docker-nexus3 project, since it already knew how to trigger the red hat build, and that's all I needed. What is a better example build? This is my first time really creating any of these, so I'm following existing examples.

Reply:
https://www.jenkins.io/doc/book/pipeline/syntax/

declarative pipelines also make these crazy if statements where you check if you build failed before continuing to the next stage go away. 

depshield has a lot of these piplelines, but if you have something that already exists and is similar to what you need then just pick that. all our 'normal' builds use declarative pipelines (i think), but they are hidden from you.

no need to change anything here, but if you ever write a pipeline from scratch i would prefer the declarative style.